src/fsharp/FSharp.Core/array.fs

```diff
@@ -278,6 +281,9 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("DistinctBy")>]
         let distinctBy projection (array:'T[]) =
+            let length = array.Length
```

Comment:
@forki   --- Null check here:  Using array without verifying if it is not null.

Reply:
just looked it up - the null check is already in outer function

Reply:
so I think it's fine

Reply:
I believe it's a public API, I'm pretty sure a null check is required,

Reply:
Yes, this is a public API. Prior to this change the null check was the first line to this function.

Reply:
oh was looking on the wrong one

Reply:
fixed. and thanks!